projects/components/src/multi-select/multi-select.component.ts

```diff
@@ -170,7 +173,17 @@ export class MultiSelectComponent<V> implements AfterContentInit, OnChanges {
   }
 
   public searchOptions(searchText: string): void {
-    this.searchSubject.next(searchText);
+    if (this.searchMode === MultiSelectSearchMode.CaseInsensitive) {
+      this.searchSubject.next(searchText);
+
+      return;
```

Comment:
this case should still emit. There's no harm in doing so - it could be useful for a component's consumer to know when the search has changed. I'd expect that from a lib component and here it's actually more complex to avoid. Also, it says so in the enum comment :)

Reply:
I missed the comment. I think this can be confusing. Until there is a known use case, I think we should not emit for this mode since filtering is applied internally. That's why i was keeping `Implicit` as it's name.

Reply:
I would prefer calling it Implicit for now and keep the emit logic as above.

Reply:
What part do you find confusing? A component is expected to emit when its state changes - the caller decides whether they care or not.

The name implicit really doesn't indicate anything to the caller - it just says that something is not directly expressed. But since we're directly expressing that, why not have it describe what it's actually doing?

Reply:
If search is happening implicitly then why emit on the search value. As a consumer am I supposed to act on it? The name `case-insensitive` doesn't convey well that there is a auto filtering applied

Reply:
As a consumer you may want to or you may not (the same as any other output). In my mind that's a much clearer name than implicit as it not only indicates a sort is occurring but how it works - if there's a clearer name though, I'm game.

Reply:
I am fine with that name, but with that name I don't think we should emit. Can we keep the same name but emit it only when there is a valid use case?

Reply:
The reason I'm caught up on this is that it's not that we're omitting something until we need it - that, I'm pretty much always for. Here, we're adding extra logic to make the behavior of an output inconsistent until we need to use it. In that case, I'd much rather have consistency even if we're not always using it. Outputs are meant to tell you about a state change, in this case, it's telling us that the search string has changed - the host is the one deciding whether to do anything with that or not.

Reply:
oh come on :)

Reply:
updated

Reply:
you're my hero!